### PR TITLE
feat(in-app-messaging): handle Carousel and Modal layouts

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes

- add `layout` value mapping for `OVERLAYS` and `MOBILE_FEED` in `AWSPinpointProvider`
- replace usage of `OVERLAYS` with 'FULL_SCREEN`
- add boilerplate and integration of `CarouselMessage` and `ModalMessage`

#### Issue #, if available

NA

#### Description of how you validated changes
Manually tested in sample app

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react-native/src/InAppMessaging/components/ModalMessage/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export { default } from './ModalMessage';
```
Comment (elorzafe) [inline packages/aws-amplify-react-native/src/InAppMessaging/components/ModalMessage/index.ts:1]: Should this have license as well?

File: packages/aws-amplify-react-native/src/InAppMessaging/components/CarouselMessage/index.ts [hunk 2]
```diff
@@ -1 +1,2 @@
+export { default } from './CarouselMessage';
```
Comment (elorzafe) [inline packages/aws-amplify-react-native/src/InAppMessaging/components/CarouselMessage/index.ts:1]: Should this have license as well?

**Review by elorzafe — APPROVED** [review]
LGTM, just curious if we need license on every file. Can you check with @sammartinez ?

File: packages/aws-amplify-react-native/src/InAppMessaging/components/ModalMessage/index.ts (hunk 1 above)
Comment (calebpollman) [reply packages/aws-amplify-react-native/src/InAppMessaging/components/ModalMessage/index.ts:1]: Will add in a follow up PR 👍🏽

File: packages/aws-amplify-react-native/src/InAppMessaging/components/CarouselMessage/index.ts (hunk 2 above)
Comment (calebpollman) [reply packages/aws-amplify-react-native/src/InAppMessaging/components/CarouselMessage/index.ts:1]: Will add in a follow up PR 👍🏽

File: packages/notifications/src/InAppMessaging/Providers/AWSPinpointProvider/utils.ts [hunk 3]
```diff
@@ -208,6 +212,30 @@ export const clearMemo = () => {
 	eventMetricsMemo = {};
 };
 
+// in the pinpoint console when a message is created with a Modal or Full Screen layout,
```
Comment (cshfang) [inline packages/notifications/src/InAppMessaging/Providers/AWSPinpointProvider/utils.ts:215]: Nit:
```suggestion
// When a message is created in the Pinpoint console with a Modal or Full Screen layout,
```

File: packages/notifications/src/InAppMessaging/Providers/AWSPinpointProvider/utils.ts [hunk 4]
```diff
@@ -208,6 +212,30 @@ export const clearMemo = () => {
...
+export const interpretLayout = (
+	layout: PinpointInAppMessage['InAppMessage']['Layout']
+): InAppMessageLayout => {
+	if (layout === PinpointInAppMessageLayout.MOBILE_FEED) {
+		return 'MODAL';
+	}
+
+	if (layout === PinpointInAppMessageLayout.OVERLAYS) {
+		return 'FULL_SCREEN';
+	}
```
Comment (cshfang) [inline packages/notifications/src/InAppMessaging/Providers/AWSPinpointProvider/utils.ts:233]: Will Pinpoint ever be deprecating the `MOBILE_FEED` and `OVERLAYS` values? This is probably not an issue if we're pinning the Pinpoint client version but these enums could just disappear overnight.

**Review by cshfang — APPROVED** [review]
Lgtm

File: packages/notifications/src/InAppMessaging/Providers/AWSPinpointProvider/utils.ts (hunk 4 above)
Comment (calebpollman) [reply packages/notifications/src/InAppMessaging/Providers/AWSPinpointProvider/utils.ts:233]: Used the enum values specifically so we would get TS errors if they disappear when we migrate to the next Pinpoint client version

**Pull request merged**